packages/eslint-plugin/docs/rules/consistent-type-assertions.mdx

```diff
@@ -115,6 +115,76 @@ const foo = <Foo props={{ bar: 1 } as Bar} />;
 </TabItem>
 </Tabs>
 
+### `arrayLiteralTypeAssertions`
+
+{/* insert option description */}
+
+Always prefer `const x: T[] = [ ... ];` to `const x = [ ... ] as T[];` (or similar with angle brackets). The rationale for this is exactly the same as for `objectLiteralTypeAssertions`.
```

Comment:
[Docs] Not everyone will read all options. Also, it's not quite exactly the same, it's just similar.

```suggestion
Always prefer `const x: T[] = [ ... ];` to `const x = [ ... ] as T[];` (or similar with angle brackets). 
The compiler will warn for excess properties of elements with this syntax, but not missing _required_ fields of those objects.
For example: `const x: { foo: number }[] = [{}]; will fail to compile, but const x = [{}] as [{ foo: number }] will succeed.
```